docs/design/image-sig-verification.md

```diff
@@ -0,0 +1,146 @@
+
+An example workflow is shown below:
+
+```sh
+                         +--------------------------------+
+                         | User creates TritonKernelCache |
+                         | (CR) with cache image reference|
+                         +---------------+----------------+
+                                         |
+                                         v
+               +-------------------------+---------------------------+
+               | Operator reads global configuration for verification|
+               +-------------------------+---------------------------+
+                                         |
+                     +-------------------+----------------------+
+                 Yes |                                          | No
+     +---------------v---------------+        +-----------------v------------------+
+     | Is signature verification     |        | Signature verification disabled    |
+     | enabled?                      |        |                                    |
+     +---------------+---------------+        +-----------------+------------------+
+                     |                                          |
+                     v                                          v
+     +---------------+---------------+        +-----------------+------------------+
+     | Trigger Cosign verification   |        | Skip verification                  |
+     +---------------+---------------+        | Mark CR as "Verified" immediately  |
+                     |                        +------------------------------------+
+            +--------+----------+
+     Success|                   | Failure
+  +---------v---------+       +-v---------------+
+  | Mark CR as        |       | Mark CR as      |
+  | "Verified"        |       | "Failed"        |
+  +---------+---------+       +-----------------+
+            |
+            v
++-----------+-----------+
+| Agents read CR status |
++-----------+-----------+
+            |
+   +--------+-----------+
+   |                    |
+Verified             Failed
+   |                    |
+   v                    v
++--+---------------+  +-+-----------------------------+
+| Run compatibility|  | Log error and halt processing |
+| checks on node   |  +-------------------------------+
++------------------+
+```
+
+### Advantages of this approach
+
+1. Centralized Verification: Performing image signature verification in the
+   control plane minimizes the computational overhead on worker nodes and
+   ensures consistent validation results.
+
+2. Dynamic Configuration: The use of a global toggle allows for seamless
+   adaptation to different security policies, ranging from strict validation
+   in production environments to relaxed validation in development clusters.
+
+### Example Cosign Verification
+
+TODO...
+
+## Open Items
+
+- Is there a (documented/best practice) pattern that openshift uses?
+- what about an admission webhook for CRDs?
```

Comment:
Leaning towards this - so the CRD doesn't get accepted unless we verify the image signature at creation time?